### PR TITLE
Update api reference contrib doc

### PR DESCRIPTION
**Pull request by kbhawkey**
See issue #14111

### CODE REVIEW

**Comment by tengqm** [comment]
/lgtm
👍

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md [hunk 1]
```diff
@@ -31,96 +38,74 @@ information, see
...
 
-The reference documentation for the Kubernetes API is generated in two separate stages:
-
-1. Generate an OpenAPI spec from the Kubernetes source code. The tools for
-this stage are at [kubernetes/kubernetes/hack](https://github.com/kubernetes/kubernetes/tree/master/hack).
-
-1. Generate an HTML file from the OpenAPI spec. The tools for this stage are at
-[kubernetes-incubator/reference-docs](https://github.com/kubernetes-incubator/reference-docs).
+```shell
+export GOPATH=$HOME/<your-local-dir>
```
[the remark this answers is not in the document]
Comment (kbhawkey) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: This directory could be any sub directory. How about substituting `$HOME/go` or `$HOME/<your-workspace>`?

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md [hunk 2]
```diff
@@ -31,96 +38,74 @@ information, see
...
 
-If you don't already have the kubernetes/website repository, get it now:
+Clone your fork of the kubernetes/website or create a local branch in `<web-base>`
+to submit the generated API reference files.
 
 ```shell
-mkdir $GOPATH/src
-cd $GOPATH/src
-go get github.com/kubernetes/website
+git clone https://github.com/<your-username>/website $GOPATH/src/github.com/<your-username>/website
```
[the remark this answers is not in the document]
Comment (kbhawkey) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: OK. I will check whether the descriptions for each repository clone.

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md [hunk 3]
```diff
@@ -176,14 +159,50 @@ git status
 The output shows the modified files:
 
 ```shell
```
[the remark this answers is not in the document]
Comment (kbhawkey) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: Sure, will check.

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (kbhawkey) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: Is the description of `<web-base>` in lines 59 - 62 clear, or should we add the comment about the name of the clone?

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: `$HOME/<workspace>` seems OK. I'm thinking about readers who want to make these docs but have no experience with Golang.

Maybe `mkdir -p $HOME/<workspace>` as well? Or:
`mkdir -p $HOME/<workspace> # make sure the directory exists`

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (sftim) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: Lines 59-62 are talking about making a clone of _kubernetes/website_ whereas this is talking about (for me) grabbig a clone of sftim/kubernetes-website. There are some subtle and important details.

**Comment by sftim** [comment]
In theory, there's a way to make this process quite a lot easier: someone with Copious Free Time[†](http://www.catb.org/~esr/jargon/html/C/copious-free-time.html) implements `make docker-update-reference-documentation`, and then it's a one line command.

I'm not convinced that's likely to happen or worthwhile right now. Nice idea though.

**Comment by sftim** [comment]
Overall, I think that the website would be better with these changes than without them.
/lgtm

**Comment by kbhawkey** [comment]
Hi, all. Let me know if these changes capture the API ref publishing instructions. My intention is to keep it simple, but clear. I have built the reference a number of times. The section that works, but the instructions offer a number of ways to do the same task, is setting up the clone of the k8s/website repo. There are a number of ways to do this -- all work equally as well, imho.

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (kbhawkey) [reply content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: I _think_ I fixed this. Let me know!

File: content/en/docs/contribute/generate-ref-docs/kubernetes-api.md [hunk 4]
```diff
@@ -175,15 +162,51 @@ git status
...
+## Locally test the documentation
+
+Build the Kubernetes documentation and verify the local preview.
+
 ```shell
-On branch master
-...
-   modified:   docs/reference/generated/kubernetes-api/v1.9/index.html
+cd <web-base>
+make docker-serve
```
Comment (sftim) [inline content/en/docs/contribute/generate-ref-docs/kubernetes-api.md]: Maybe mention that the reader should look at http://localhost:1313/ and review the changes?
It's also OK to infer that if they got this far they already know to do that.

**Comment by kbhawkey** [comment]
Preview page:
https://deploy-preview-15114--kubernetes-io-master-staging.netlify.com/docs/contribute/generate-ref-docs/kubernetes-api/

**Pull request merged**